web_src/js/features/repo-issue.js

```diff
@@ -680,3 +680,14 @@ export function initIssueTemplateCommentEditors($commentForm) {
     initCombo($(el));
   }
 }
+
+export function initArchivedLabelHandler() {
```

Comment:
Why this function is needed? IMO the template should output correct HTML code.

Reply:
We are not handling the at the server side, if we want to handle it server side then we need to count the archived labels and send it to template and having a count check that will show and hide the checkbox, so same we have done with client side we are counting the `data-is-archived` and if `len == 0` hide the checkbox and it working that way , we don't have to put extra query burden on backend side.

Reply:
I do not see why "count the archived labels".

If I understand correctly, the code is doing: for `data-is-archived` label, hide it if it doesn't contains `checked` class, no "count".

Reply:
We have the below checkbox which we hide if there is no `archived` label, and `checked` property is not relevent. 
Simply if there is  an archived label in the list the show the checkbox other wise hide it. That we are evaulating via 
`data-is-archived` attribute.

![image](https://github.com/go-gitea/gitea/assets/80308335/3e8175f4-9ce5-4334-95fc-90700f42fd0d)

Reply:
I still don't understand. Read the code line by line:

```
export function initArchivedLabelHandler() {
  const archivedLabelEl = document.querySelector('.archived-label-hint'); // query '.archived-label-hint'
  if (!archivedLabelEl) {
    return;  // if no '.archived-label-hint', return
  }

  // query all labels with '[data-is-archived]'
  const archivedLabels = document.querySelectorAll('[data-is-archived]');
  for (const label of archivedLabels) {
    // for each '[data-is-archived]' label, hide it if it is not checked.
    toggleElem(label, label.classList.contains('checked'));
  }
}
```

I do not see the logic for `We have the below checkbox which we hide if there is no archived label, and checked property is not relevent.` here.

Reply:
Sorry for the confussion.
Here I have explained
https://github.com/go-gitea/gitea/pull/27451/files#r1349952100

Reply:
> Sorry for the confusion. Here I have explained https://github.com/go-gitea/gitea/pull/27451/files#r1349952100

> This is an another case on where there is no checkbox but we are showing archived label if that label is selected in the list and hence if that anchor tag contains checked class other wise we will hide.

But for this case, I think the complete HTML could still be generated by backend template without JS? eg: `{{if and .IsArchived (not .Checked)}}gt-hidden{{end}}`.

The JS code seems fragile, I still have difficulty to understand the necessity.

Reply:
Well… the moment we want to dynamically show/hide archived labels (and we want that without a page reload), we have no choice but to show/hide it with JS.

While I agree that you should only use JS where necessary, I'd argue this is one of the places where it makes sense to defer the task to JS.

Reply:
> Well… the moment we want to dynamically show/hide archived labels (and we want that without a page reload), we have no choice but to show/hide it with JS.

I still do not understand the point. `initArchivedLabelHandler` is only called in `onDomReady`, and is only called once, it only does the logic above (https://github.com/go-gitea/gitea/pull/27451/files#r1349870436 : for data-is-archived label, hide it if it doesn't contains checked class, no "count")

> this is one of the places where it makes sense to defer the task to JS.

I still have the difficulty to understand why it makes sense ....

----

So, maybe I misunderstood something, no blocker from my side if it is the right thing to do.

Reply:
> I still don't understand. Read the code line by line:
> 
> ```
> export function initArchivedLabelHandler() {
>   const archivedLabelEl = document.querySelector('.archived-label-hint'); // query '.archived-label-hint'
>   if (!archivedLabelEl) {
>     return;  // if no '.archived-label-hint', return
>   }
> 
>   // query all labels with '[data-is-archived]'
>   const archivedLabels = document.querySelectorAll('[data-is-archived]');
>   for (const label of archivedLabels) {
>     // for each '[data-is-archived]' label, hide it if it is not checked.
>     toggleElem(label, label.classList.contains('checked'));
>   }
> }
> ```
> 
> I do not see the logic for `We have the below checkbox which we hide if there is no archived label, and checked property is not relevent.` here.

Let me re-iterate again 

There are two cases where we are using this archived functionality 
1 :- issue list /pr list page
2:- Any particular issue or pr detail page


For 2 .  The above code is used in particular issue or pr detail page's label selector click, 
![image](https://github.com/go-gitea/gitea/assets/80308335/1af64707-c593-4e7d-9da0-b52378c50f91)

here is a global init function `initArchivedLabelHandler` is called which is handled ther show/hide archived labels

Like below label is marked `archived` is showed
![image](https://github.com/go-gitea/gitea/assets/80308335/1321010f-2ede-4c37-9549-97b8e9f4976c)

So conclusion when any pr or issue is marked with archived label we will show those label in the drop down  list other wise we hide.



For 1 .  For issue list or pr list page :- 
![image](https://github.com/go-gitea/gitea/assets/80308335/8118007b-a904-4657-a48f-07874a86bb7c)
![image](https://github.com/go-gitea/gitea/assets/80308335/240c260d-9cb2-4e4a-966e-9df70a5e0f43)

if we have any label in the list which is archived we are showing the checkbox, and if there is no archived label then we are hiding that checkbox with the init function `initArchivedLabelFilter` which supplies `archived=true` in the query param for server side handling and here we have a click listener registered on the checkbox which will render the archived labels.


These two cases are different.